backend/src/main/java/com/bakdata/conquery/models/index/MapInternToExternMapper.java

```diff
@@ -67,14 +77,31 @@ public class MapInternToExternMapper extends NamedImpl<InternToExternMapperId> i
 	@Getter(onMethod_ = {@TestOnly})
 	private MapIndex int2ext = null;
 
-
 	@Override
-	public synchronized void init() {
+	public synchronized MapInternToExternMapper init() {
+
+		// An alternative would be to register the converter programmatically for the internal mappers but not for the
+		// test mapper.
+		if (mapIndex == null && config == null) {
+			log.trace("Injections were null. Skipping init, because class was deserialized by a test object mapper");
+			return this;
+		}
+
+		if (dataset == null) {
+			// TODO remove this as the dataset can be derived from the storage and does not need to be explicitly declared
+			dataset = storage.getDataset();
+		}
 
 		final URI resolvedURI = FileUtil.getResolvedUri(config.getIndex().getBaseUrl(), csv);
 		log.trace("Resolved mapping reference csv url '{}': {}", this.getId(), resolvedURI);
 
-		int2ext = mapIndex.getIndex(new MapIndexKey(resolvedURI, internalColumn, externalTemplate));
+		MapIndexKey key = new MapIndexKey(resolvedURI, internalColumn, externalTemplate);
+		try {
+			int2ext = mapIndex.getIndex(key);
```

Comment:
das heißt die werden jetzt schon beim reinladen initialisiert? Das halte ich für keine gute Idee

Reply:
Dadurch wird der Upload endlos verlangsamt

Reply:
Ich kann das nach hinten verlagern, aber das parsen der Mappings ist normalerweise sehr fix.

Reply:
nicht wenn man es seriell macht was ja dadurch implizit passiert

Reply:
Ich probiere es mal mit einem wrap in ComputableFuture